dev/system-test/firestore.ts

```diff
@@ -222,11 +222,10 @@ describe('CollectionReference class', () => {
   });
 
   it('supports withConverter()', async () => {
-    const ref = firestore
+    const ref = await firestore
       .collection('col')
       .withConverter(postConverter)
-      .doc('doc');
-    await ref.set(new Post('post', 'author'));
+      .add(new Post('post', 'author'));
```

Comment:
Please also add a corresponding unit test. Ideally, our entire SDK surface should be tested through unit tests first and integration tests second.